server/spec/requests/api/v1/api_users_spec.rb

```diff
@@ -120,9 +154,46 @@
           it 'returns same params as entered' do
             expect(json['first_name'],json['last_name'],json['city'], json['email'], json['password_digest'],json['mentor'],json['mentee']).to eq('Bobby', 'Dylan','Mexico', 'Test_email@email.com','password1', True, False)
           end
+        end
+
+
+
+        context 'when email already in use' do
+          let(:api_user_id) {api_users.first.id}
+          let(:invalid_email_update) do
+            # send json payload
+            { 'email': 'Test_email@email.com'}.to_json
+            before { patch "/api/v1/api_users/#{api_user_id}/", params: invalid_email_updates}
+
+            it 'returns status code 422' do
+              expect(response).to have_http_status(422)
+            end
+
+            it 'returns message informing no user with that id' do
+              expect(json['message']).to match(/Validation failed: Email has already been taken/)
+            end
+          end
+        end
 
+        context 'when username already in use' do
+          let(:api_user_id) {api_users.first.id}
+          let(:invalid_username_update) do
+            # send json payload
+            { 'username': 'user11'}.to_json
```

Comment:
To clarify: I meant that the `let` which "opened a block" with its `do` needs to "close its block" with an `end` just after the value that `invalid_username_update` should have.

```suggestion
            { 'username': 'user11'}.to_json
          end
```

Reply:
Here's a link to Rubymonstas, about Blocks: http://ruby-for-beginners.rubymonstas.org/blocks.html

And here, Satish explains blocks in other words: http://rubylearning.com/satishtalim/ruby_blocks.html

Reply:
This let block is closed just before the context block is closed. Is there a reason why it should be closed so early, the invalid attributes are used for the subsequent two tests before the let is closed.

Reply:
I looked into this further and have made the changes